lib/sync/sync-processor.js

```diff
@@ -52,7 +52,8 @@ function recordProcessTime(startTime, success) {
 function markDatasetClientAsCompleted(datasetClientId, startTime, callback) {
   syncStorage.updateDatasetClient(datasetClientId, {syncLoopEnd: Date.now(), syncCompleted: true, syncScheduled: null}, function(err){
     if (err) {
-      syncUtil.doLog(datasetId, "error", "Error when update dataset client with id " + datasetClientId + ". error = " + util.inspect(err));
+      // TODO: datasetId here isn't defined, is it?
```

Comment:
@grdryn I think here, if we have no access to the datasetId, we should be using `syncUtil.SYNC_LOGGER`. The first argument is meant to define the dataset to log for, any other place where there is no access to a dataset id seems to use `syncUtil.SYNC_LOGGER`.

If we could change this function to accept a `DatasetClient` object we could use the `getId()` and `getDatasetId()` methods to make this complete. 

I haven't looked much into it but I'm not sure that this is possible though without calling `doReadDatasetClient` which would have to call out to MongoDB, which is probably a bit overkill just to provide a log.